public/coffee/ide/editor/directives/aceEditor.coffee

```diff
@@ -374,6 +374,18 @@ define [
 					if scope.eventsBridge?
 						session.on "changeScrollTop", onScroll
 
+					hasLintingError = null
+					session.on('changeAnnotation', () ->
+						hasErrors = session
```

Comment:
Will this not all pick up errors from the compile logs (which are injected into the editor as annotations as well)?

Reply:
As discussed, I've added an extra flag to linting output (https://github.com/sharelatex/ace/pull/4), which we then filter on when listening for changes